algorithms/src/Kokkos_Random.hpp

```diff
@@ -886,29 +886,27 @@ class Random_XorShift64_Pool {
  public:
   using generator_type = Random_XorShift64<DeviceType>;
 
-  KOKKOS_INLINE_FUNCTION
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  KOKKOS_FUNCTION
   Random_XorShift64_Pool() {
-    num_states_ = 0;
-    padding_    = 0;
+    KOKKOS_IF_ON_DEVICE((
+        /* nonsensical initialization in the name of backward compatibility */
+        num_states_ = 0; padding_ = 0;))
+    KOKKOS_IF_ON_HOST(init(0, execution_space().concurrency());)
   }
+
   Random_XorShift64_Pool(uint64_t seed) {
     num_states_ = 0;
 
     init(seed, execution_space().concurrency());
   }
+#else
+  Random_XorShift64_Pool(uint64_t seed = 0) {
```

Comment:
should this be marked explicit?

Reply:
See my comment above, but yes in general I would declare such constructor with the function specifier explicit.

Reply:
I opted for not adding any explicit function specifier.  I don't think it is worth the trouble.